ANNOUNCEMENT.md

```diff
@@ -0,0 +1,284 @@
+exclusively in Rust. This venture aimed not only to comprehend the language but also
+to evaluate whether Rust truly lived up to its promises. The resounding answer is a
+definitive "yes," and more.
+
+My day job involves certifying iceoryx written in C++ for use in safety-critical
+ASIL-D environments. While grappling with algorithmic challenges, the burden of C++
+introduces a myriad of additional complications, consuming considerable time and
+causing frustration.
+
+* **Documentation:** Managing doxygen code examples in harmony with the codebase.
+* **Lifetimes:** Navigating lifetime dependencies when working with RAII and factories.
+* **Concurrency:** Preventing inadvertent multi-threaded usage and certifying
+    concurrent and lock-free code.
+* **Build-System:** Handling dependencies and third-party packages seamlessly.
+* **Templates:** Certifying generic C++ code for various but valid types.
+* **Static Code Analysis:** Choosing the right tools and adhering to Misra,
+    Autosar, C++ Core Guidelines, or a combination.
+
+Why dwell in the abstract when we can scrutinize a real-world example, comparing
+iceoryx (C++) to iceoryx2 (Rust)? Brace yourself for a journey into a future where
+Rust takes the lead in inter-process zero-copy communication within our specialized
+domain. Iceoryx2 is not just an upgrade—it's a leap forward into a safer, more
+efficient era.
+
+## Example: Sending Data
+
+In both iterations of iceoryx, the communication revolves around ports as endpoints. 
+In this context, the `Publisher` assumes the role of the sender in a publish-subscribe 
+messaging pattern. When users intend to transmit data, they typically follow a set
+of common steps across both versions.
+
+1. Invoke `Publisher::loan()` to obtain a `Sample` for storing the data to be sent.
+2. Utilize `Publisher::send(sample)` to dispatch the data to all receiving endpoints (`Subscriber`).
+
+Represented here is a simplified interface for this `Publisher`:
+
+```cpp
+// C++
+class Publisher {
+  public:
+    Sample loan();
+    void send(Sample &&sample);
+};
+```
+
+```rust
+// Rust
+struct Publisher {}
+
+impl Publisher {
+    fn loan(&'publisher self) -> Sample<'publisher> {}
+    fn send(&self, sample: Sample) {}
+}
+```
+
+## Documentation Best Practices: C++ vs. Rust
+
+When it comes to documenting code and ensuring the accuracy of code examples, both C++
+and Rust offer distinct approaches. Let's delve into the practices and tools used in
+each language:
+
+### C++
+
+In the C++ realm, internal documentation is commonly managed using Doxygen, a tool
+providing `@code`/`@endcode` tags for embedding code examples. However, there's a
+crucial limitation—Doxygen does not verify whether the code is running or compiling.
+This introduces a potential risk, as refactoring in one part of the codebase may
+inadvertently break examples located elsewhere.
+
+```cpp
+/// @code
+///   std::cout << "hello world, without include iostream";
+/// @endcode
```

Comment:
We don't make use of it in iceoryx but there is an option to embed code snippets from a C++ file. This prevents the issue with non-compiling examples. But one needs to look at the doxygen output to check whether the examples is still semantically correct. With Rust it's just at the start of the method and easier to check in PRs